bftengine/src/bftengine/ReplicaImp.hpp

```diff
@@ -62,34 +60,31 @@ class StateTransferMsg;
 class ReplicaStatusMsg;
 class ReplicaImp;
 
+using bftEngine::ReplicaConfig;
 using PtrToMetaMsgHandler = void (ReplicaImp::*)(MessageBase* msg);
 
 class ReplicaImp : public InternalReplicaApi, public IReplicaForStateTransfer {
  protected:
-  // system params
-  const ReplicaId myReplicaId;
-  const uint16_t fVal;
-  const uint16_t cVal;
+  ReplicaConfig config_;
+
   const uint16_t numOfReplicas;
-  const uint16_t numOfClientProxies;
   const bool viewChangeProtocolEnabled;
-  const bool supportDirectProofs;  // TODO(GG): add support
-  const bool debugStatisticsEnabled;
+  const bool supportDirectProofs = false;  // TODO(GG): add support
 
   // pointers to message handlers
   const std::unordered_map<uint16_t, PtrToMetaMsgHandler> metaMsgHandlers;
 
   // communication
-  class MsgReceiver;  // forward declaration
-  IncomingMsgsStorage incomingMsgsStorage;
-  MsgReceiver* msgReceiver;
+  class MsgReceiver;                                // forward declaration
+  IncomingMsgsStorage incomingMsgsStorage = 20000;  // TODO(GG): use configuration
```

Comment:
This is a bit strange. You are assigning an integer to a class instance. I assume it's using that in the constructor somehow, but it's not apparent from that line. Maybe change to `incomingMsgsStorage(MAX_INCOMING_MSGS)`?

Reply:
yes, but this syntax doesn't work